File: app/src/main/java/me/zhanghai/android/files/provider/common/PosixUser.java
/*
 * Copyright (c) 2018 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.common;

import android.os.Parcel;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java8.nio.file.attribute.UserPrincipal;

public class PosixUser extends PosixPrincipal implements UserPrincipal {

    public PosixUser(int id, @Nullable ByteString name) {
        super(id, name);
    }


    public static final Creator<PosixUser> CREATOR = new Creator<PosixUser>() {
        @NonNull
        @Override
        public PosixUser createFromParcel(@NonNull Parcel source) {
            return new PosixUser(source);
        }
        @NonNull
        @Override
        public PosixUser[] newArray(int size) {
            return new PosixUser[size];
        }
    };

    protected PosixUser(@NonNull Parcel in) {
        super(in);
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }
}

File: app/src/main/java/me/zhanghai/android/files/provider/common/PosixUser.kt
/*
 * Copyright (c) 2018 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.common

import android.os.Parcel
import android.os.Parcelable
import java8.nio.file.attribute.UserPrincipal

class PosixUser : PosixPrincipal, UserPrincipal {
    constructor(id: Int, name: ByteString?) : super(id, name)

    private constructor(source: Parcel, loader: ClassLoader?) : super(source, loader)

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.ClassLoaderCreator<PosixUser> {
            override fun createFromParcel(source: Parcel): PosixUser =
                createFromParcel(source, null)

            override fun createFromParcel(source: Parcel, loader: ClassLoader?): PosixUser =
                PosixUser(source, loader)

            override fun newArray(size: Int): Array<PosixUser?> = arrayOfNulls(size)
        }
    }
}
